lib/ae_mdw_web/websocket/subscriptions.ex

```diff
@@ -130,40 +134,48 @@ defmodule AeMdwWeb.Websocket.Subscriptions do
   defp unsubscribe_all(pid) do
     @subs_pid_channel
     |> :ets.match_object({pid, :"$1"})
-    |> Enum.each(fn {^pid, channel} ->
-      :ets.delete_object(@subs_pid_channel, {pid, channel})
-      {:ok, channel_key} = channel_key(channel)
+    |> Enum.each(fn {^pid, channel_key} ->
+      :ets.delete_object(@subs_pid_channel, {pid, channel_key})
       :ets.delete_object(@subs_channel_pid, {channel_key, pid})
     end)
   end
 
-  defp validate_subscribe(pid, channel) do
-    with {:ok, channel_key} <- channel_key(channel),
-         {:exists, false} <- {:exists, exists?(pid, channel)} do
+  defp validate_subscribe(pid, version, channel) do
+    with {:ok, channel_key} <- channel_key(version, channel),
+         {:match, :none} <- {:match, match(pid, channel_key)} do
       {:ok, channel_key}
     else
       {:error, _reason} -> {:error, :invalid_channel}
-      {:exists, true} -> {:error, :already_subscribed}
+      {:match, {:ok, _channel}} -> {:error, :already_subscribed}
     end
   end
 
-  defp validate_unsubscribe(pid, channel) do
-    with {:ok, channel_key} <- channel_key(channel),
-         true <- exists?(pid, channel) do
-      {:ok, channel_key}
+  defp validate_unsubscribe(pid, version, channel) do
+    with {:ok, channel_key} <- channel_key(version, channel),
+         {:ok, channel} <- match(pid, channel_key) do
```

Comment:
The `channel` returned by `match` is exactly the same value as the input of the unsubscribe. So since it's already available in the function the `exists?` function is enough (keeping the code simpler).

Reply:
It might not be the same because right now the subscribe/unsubscribe allows you to subscribe to a channel like `ak_wTPFpksUJFjjntonTvwK4LJvDw11DPma7kZBneKbumb8yPeFq` and unsubscribe  to `ok_wTPFpksUJFjjntonTvwK4LJvDw11DPma7kZBneKbumb8yPeFq`. Same address, different type

Reply:
It would be the same because when there is a subscription for `ak_wTPFpksUJFjjntonTvwK4LJvDw11DPma7kZBneKbumb8yPeFq` it is an alias for the binary that is matched with the value in the transactions (that is not the encoded pubkey). In this sense we might consider that `ak_wTPFpksUJFjjntonTvwK4LJvDw11DPma7kZBneKbumb8yPeFq` and `ok_wTPFpksUJFjjntonTvwK4LJvDw11DPma7kZBneKbumb8yPeFq` would be aliases for same subscription.

Reply:
Exactly, but in order to remove the value from ets we need to know the exact address in which the user was subscribed to

Reply:
The address is actual the binary which is the same for `ak_wTPFpksUJFjjntonTvwK4LJvDw11DPma7kZBneKbumb8yPeFq` and `ok_wTPFpksUJFjjntonTvwK4LJvDw11DPma7kZBneKbumb8yPeFq`

Reply:
The address is actuallly the binary which is the same for `ak_wTPFpksUJFjjntonTvwK4LJvDw11DPma7kZBneKbumb8yPeFq` and `ok_wTPFpksUJFjjntonTvwK4LJvDw11DPma7kZBneKbumb8yPeFq`